agent/src/main.rs

```diff
@@ -57,6 +55,22 @@ async fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>
         run_metrics_server().await.unwrap();
     }));
 
+    let discovery_handler_map = Arc::new(Mutex::new(HashMap::new()));
+    let discovery_handler_map_clone = discovery_handler_map.clone();
+    let (new_discovery_handler_sender, _): (
+        broadcast::Sender<String>,
```

Comment:
Should we use something more typed, if nothing just an alias again with some comment explaining the payload?

Reply:
added in fa4eba408f57f2